lib/grpc-lwt/client.ml

```diff
@@ -78,14 +71,13 @@ module Rpc = struct
     [ `write ] H2.Body.t -> [ `read ] H2.Body.t Lwt.t -> 'a Lwt.t
 
   let bidirectional_streaming ~f write_body read_body =
-    let decoder_stream, decoder_push = Lwt_stream.create () in
-    Lwt.async (fun () ->
-        let+ read_body in
-        Connection.grpc_recv_streaming read_body decoder_push);
     let encoder_stream, encoder_push = Lwt_stream.create () in
-    Lwt.async (fun () ->
-        Connection.grpc_send_streaming_client write_body encoder_stream);
-    f encoder_push decoder_stream
+    let decoder_stream, decoder_push = Lwt_stream.create () in
+    let res = f encoder_push decoder_stream in
+    let* () = Connection.grpc_send_streaming_client write_body encoder_stream in
+    let* read_body = read_body in
+    Connection.grpc_recv_streaming read_body decoder_push;
```

Comment:
Is the actual connection stopped when you cancel the promise?

Reply:
If you mean the H2 connection, no, that stays untouched.

Reply:
Actually, let me do some more tests. Because the error, that I saw before, disappeared with this change, so I don't know the behaviour when it gets cancelled.

Reply:
Ok, I stress-tested with 50 concurrent threads and some artificial errors, and everything works nicely, when the `call` promise gets cancelled.